src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -143,6 +164,7 @@ result<http::client::request_header> request_creator::make_head_object_request(
     if (ec) {
         return ec;
     }
+    url_encode_target(header);
```

Comment:
did you consider putting this in `make_target`?

Reply:
This has to be done post `add_auth` as it would result in double encoding when creating canonical request form for signature calculation.

Reply:
But is the encoding that happens in `split_target` needed at all if we already have it encoded in the URL?